src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java

```diff
@@ -509,11 +516,12 @@ private AssigningResult(final IRequestResolver resolver, final List<IToken<T>> c
         }
 
         @Override
-        public int compareTo(@NotNull final AssigningResult o)
+        public int compareTo(@NotNull final AssigningResult<T> o)
         {
             return this.children.size() != o.children.size() ? this.children.size() - o.children.size() : o.getResolver().getPriority() - this.getResolver().getPriority();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS864)